src/lib/types.ts

```diff
@@ -10,6 +10,7 @@ export type Transaction = {
   value: bigint;
   gasPrice?: bigint;
   timestamp: bigint;
+  isContractInteraction?: boolean;
```

Comment:
You can remove this, we'll stick with the transaction interface provided by JSON-RPC which doesn't involve an `isContractInteraction` property.